src/find/matchers/exec.rs

```diff
@@ -13,6 +14,27 @@ use std::process::Command;
 
 use super::{Matcher, MatcherIO, WalkEntry};
 
+fn check_path_integrity() -> Result<(), Box<dyn Error>> {
+    let path_dirs = env::var("PATH")?;
+    for dir_entry in env::split_paths(&path_dirs) {
+        // We can securely unwrap (or expect) the value of dir_entry string
+        // conversion on message error cause the env::var returns an VarError
+        // variant that indicates if the variable (in this case PATH) contains
+        // invalid Unicode data.
+        let dir_entry_str = dir_entry.to_str().expect("Unexpected conversion error");
+        if !dir_entry.is_absolute() || dir_entry.is_file() || dir_entry_str.is_empty() {
```

Comment:
```suggestion
        if !dir_entry.is_absolute() || dir_entry.as_os_str().is_empty() {
```

I don't think the `is_file()` check is necessary, GNU find doesn't do it.

Reply:
Ok